common/raw_hashtable.h

```diff
@@ -853,6 +899,7 @@ auto BaseImpl<InputKeyT, InputValueT, InputKeyContextT>::InsertImpl(
     LookupKeyT lookup_key, KeyContextT key_context)
     -> std::pair<EntryT*, bool> {
   CARBON_DCHECK(alloc_size() > 0);
+  PrefetchStorage();
```

Comment:
I wondered if PrefetchMetadata() would explain this a bit more clearly, since it's the metadata pointer that is being used, and presumably it is only metadata reads that are going to benefit?

Reply:
Internally, it uses the storage pointer?

But generally, there is a contiguous block of memory for 'storage' -- first the metadata and then the entries. Because the metadata comes first, it definitely benefits the most from this.

It's not clear how far the prefetch helps -- is it more than a cache line? I don't really know, and I think CPU docs around this are intentionally vague to allow pretty wide ranging implementations.

I also have a hard time naming this because, as discussed in the PR description, it doesn't seem to really work well in practice. That makes me uncertain how to name it.

But the real goal here is to provide hooks that we could choose to do arbitrary prefetching within, so maybe that speaks to better names reflecting where these hooks are placed rather than what memory would be fetched? The second one is actually nearly that already... So maybe `PrefetchForStart` and `PrefetchForEntryGroup`? Maybe there are better names?

Reply:
Switching to `PrefetchMetadata` for now and landing with the approval. Not sure it's worth holding this up on better names. Happy to rename with any suggestions in follow-up PRs.

Reply:
Sounds good, thanks :)